src/CheckBox/index.jsx

```diff
@@ -36,8 +36,8 @@ class CheckBox extends React.Component {
         checked: true,
       });
     }
-    if (this.onChange) {
-      this.onChange();
+    if (this.onChangeState) {
+      this.onChangeState();
```

Comment:
Take a look at `asInput`'s `handleChange` function. It fires whatever `onChange` function has been passed to its descendant component, so it should handle this for you automatically. What happens if you change your parameter name back to `onChange` and remove this `onChangeState` call from the checkbox component?